server/mqtt.go

```diff
@@ -4267,8 +4267,9 @@ func (s *Server) mqttProcessPubRel(c *client, pi uint16, trace bool) error {
 func (c *client) mqttHandlePubRetain() {
 	pp := c.mqtt.pp
 	isRetained := mqttIsRetained(pp.flags)
-	isSparkbBirth, typ, groupID, nodeID, deviceID := sparkbIsBirth(pp.subject, sparkbNamespacePrefix)
-	if !isRetained && !isSparkbBirth {
+	isBirth, _, isCertificate := sparkbParseBirthDeathTopic(pp.topic)
```

Comment:
My original question was, are those spark messages meant to have the retain flag when they are received by the server. If they MUST not, then we should probably check that here and fail if that is the case?

Reply:
_[tck-id-topics-nbirth-mqtt] NBIRTH messages MUST be published with MQTT QoS equal to 0
and retain equal to false._

I'll make the change to treat it as an error, even though I am 0/5 on whether we should be enforcing these NODE and DEVICE aspects of the spec - we don't fail for an invalid the QoS for example.